client/chat/actions.ts

```diff
@@ -193,89 +196,73 @@ export interface DeactivateChannel {
   }
 }
 
+/**
+ * A type that allows using the socket events directly as payload for dispatch actions, by including
+ * the channel in the payload, since the channel is not included in the event itself (it is part of
+ * the websocket URL).
+ */
+type PayloadWithChannel<T = void> = { channel: string } & Omit<T, 'action'>
```

Comment:
This is a very weird type imo. If you're going to omit things from a generic type, it should be limited to types that actually contain the key you're omitting

I would also recommend not using default values for generics unless they are actually useful. The default type here does not seem useful/good, makes it easy to misuse the type, and prevents better type inference in places that might be utilizing this type